widgets/deploymentsView/src/columns.tsx

```diff
@@ -28,7 +28,42 @@ export interface DeploymentsViewColumnDefinition {
     render(deployment: Deployment): ReactNode;
 }
 
-const i18nPrefix = 'widgets.deploymentsView.columns';
+const i18nPrefix = 'widgets.deploymentsView';
+const i18nColumnsPrefix = `${i18nPrefix}.columns`;
+
+interface IconDescription {
+    labelI18nKey: string;
+    props: IconProps;
+}
+/**
+ * A CID (Constrained Identity Function)
+ * See https://kentcdodds.com/blog/how-to-write-a-constrained-identity-function-in-typescript
+ */
+const createIconDescriptions = <T extends Record<string, IconDescription>>(iconDescriptions: T) => iconDescriptions;
+
+const statusIconDescriptions = createIconDescriptions({
+    inProgress: {
+        labelI18nKey: 'inProgress',
+        props: { name: 'spinner', color: 'orange' }
+    },
+    requiresAttention: {
+        labelI18nKey: 'requiresAttention',
+        props: { name: 'exclamation', color: 'red' }
+    }
+});
+type StatusIconName = keyof typeof statusIconDescriptions;
+const renderStatusIcon = (iconName: StatusIconName) => {
+    const { Icon, Popup } = Stage.Basic;
+    const iconDescription = statusIconDescriptions[iconName];
+    const label = Stage.i18n.t(`${i18nPrefix}.iconLabels.${iconDescription.labelI18nKey}`);
+
+    return (
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        <Popup trigger={<Icon aria-label={label} {...iconDescription.props} />} position="top center">
```

Comment:
If you simplify the code (see my first comment), you should be able to change it to:
```suggestion
        <Popup trigger={<Icon aria-label={label} {...iconDescription} />} position="top center">
```

I would consider changing `iconDescription` to have better name, probably `iconProps` is more accurate.

Reply:
I've applied both suggestions :slightly_smiling_face: